### PR TITLE
Allow instantiating random walk classes with pre-defined hyper-parameters

### PR DESCRIPTION
**Pull request by kjun9**
This updates the `__init__` and `run` methods for the random walker classes:
* `UniformRandomWalk`
* `BiasedRandomWalk`
* `UniformRandomMetaPathWalk`
* `TemporalRandomWalk`

which are all supposed to be suitable for obtaining context pairs for unsupervised learning. This refactoring would allow users to instantiate a walker object with its hyperparameters pre-defined.

If this seems reasonable, I was hoping to do a bit more follow-up to make #919 and #536 possible:
* define a separate class for random walkers instead of `GraphWalk` (which currently also contains BFS-type walkers for the SAGE algorithms) - maybe just `RandomWalk`?
* update `UnsupervisedSampler` to optionally take a `RandomWalk` object as a parameter.

Thoughts?

See #1178

### CODE REVIEW

File: stellargraph/data/explorer.py [hunk 1]
```diff
@@ -38,6 +38,15 @@
 from ..random import random_state
 
 
+def _default_if_none(value, default, name, ensure_not_none=True):
+    value = value if value is not None else default
+    if ensure_not_none and value is None:
+        raise ValueError(
+            f"{name}: expected a value to be specified in either `__init__` or `run`, found: None."
```
Comment (huonw) [inline stellargraph/data/explorer.py]: Minor tweak, potentially:

```suggestion
            f"{name}: expected a value to be specified in either `__init__` or `run`, found None in both"
```

File: stellargraph/data/explorer.py [hunk 2]
```diff
@@ -179,22 +188,38 @@ def _check_sizes(self, n_size):
 class UniformRandomWalk(GraphWalk):
     """
     Performs uniform random walks on the given graph
+
+    Args:
+        graph (StellarGraph): Graph to traverse
+        graph_schema (GraphSchema, optional): Graph schema
```
Comment (huonw) [inline stellargraph/data/explorer.py]: I think none of these classes use `graph_schema` at all, so we could potentially not accept the argument in each of them (theoretically we would have `GraphWalk` not call `create_graph_schema` too, which I guess could happen as part of splitting out `RandomWalk`). That is

```suggestion
```

and
```python
def __init__(self, graph, n=None, length=None, seed=None):
    super().__init__(graph, graph_schema=None, seed=seed) # or super().__init__(graph, seed=seed)
```

and similarly for each of the other classes.

Thoughts?

**Comment by kjun9** [comment]
> because those want to pass around an object that can just have .run(start_nodes) called without also having to pass through all the possible arguments for each run

Yep! As it stands, it won't work for `TemporalRandomWalk.run` since its parameters are a bit different, so would have to think of something

File: stellargraph/data/explorer.py (hunk 2 above)
Comment (kjun9) [reply stellargraph/data/explorer.py]: Makes sense, I'll update them

**Pull request merged**